File: apps/Camera-Vitals/Android/app/src/main/java/ai/zetic/demo/cameravitals/ui/Theme.kt
package ai.zetic.demo.cameravitals.ui

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.lightColorScheme
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp

/** Light medical-clean design tokens (mirrors iOS Theme). */
object Theme {
    val background = Color(0xFFF5F7FC)
    val card = Color.White
    val accent = Color(0xFF2E73F2)
    val accentSoft = Color(0x1F2E73F2)
    val textPrimary = Color(0xFF1A2133)
    val textSecondary = Color(0xFF6B788C)
    val good = Color(0xFF33B87A)
    val fair = Color(0xFFF29E2E)
    val poor = Color(0xFFE64D4D)

    val cardRadius = 22.dp

    fun quality(q: Double): Color = if (q >= 0.6) good else if (q >= 0.3) fair else poor
    fun qualityLabel(q: Double): String =
        if (q >= 0.6) "Good signal" else if (q >= 0.3) "Fair signal" else "Stabilizing…"
}

@Composable
fun CameraVitalsTheme(content: @Composable () -> Unit) {
    MaterialTheme(
        colorScheme = lightColorScheme(
            primary = Theme.accent,
            background = Theme.background,
            surface = Theme.card,
            onPrimary = Color.White,
            onBackground = Theme.textPrimary,
            onSurface = Theme.textPrimary
        ),
        content = content
    )
}

/** Reusable soft white card. */
@Composable
fun Card(modifier: Modifier = Modifier, content: @Composable () -> Unit) {
    Box(
        modifier
            .shadow(8.dp, RoundedCornerShape(Theme.cardRadius), clip = false)
            .clip(RoundedCornerShape(Theme.cardRadius))
            .background(Theme.card)
    ) { content() }
}

File: apps/Camera-Vitals/iOS/CameraVitals/App/Theme.swift
import SwiftUI

/// Light medical-clean design tokens.
enum Theme {
    static let background = Color(red: 0.96, green: 0.97, blue: 0.99)   // soft off-white
    static let card = Color.white
    static let accent = Color(red: 0.18, green: 0.45, blue: 0.95)       // calm clinical blue
    static let accentSoft = Color(red: 0.18, green: 0.45, blue: 0.95).opacity(0.12)
    static let textPrimary = Color(red: 0.10, green: 0.13, blue: 0.20)
    static let textSecondary = Color(red: 0.42, green: 0.47, blue: 0.55)
    static let good = Color(red: 0.20, green: 0.72, blue: 0.48)
    static let fair = Color(red: 0.95, green: 0.62, blue: 0.18)
    static let poor = Color(red: 0.90, green: 0.30, blue: 0.30)

    static let cardRadius: CGFloat = 22
    static let cardShadow = Color.black.opacity(0.06)

    static func quality(_ q: Double) -> Color {
        if q >= 0.6 { return good }
        if q >= 0.3 { return fair }
        return poor
    }

    static func qualityLabel(_ q: Double) -> String {
        if q >= 0.6 { return "Good signal" }
        if q >= 0.3 { return "Fair signal" }
        return "Stabilizing…"
    }
}

/// Reusable soft card container.
struct Card<Content: View>: View {
    @ViewBuilder var content: Content
    var body: some View {
        content
            .background(Theme.card)
            .clipShape(RoundedRectangle(cornerRadius: Theme.cardRadius, style: .continuous))
            .shadow(color: Theme.cardShadow, radius: 12, x: 0, y: 6)
    }
}
